src/main/java/htsjdk/samtools/BAMRecordCodec.java

```diff
@@ -174,6 +184,15 @@ public void encode(final SAMRecord alignment) {
         }
     }
 
+    private int getUshort(int value) {
+        return value & (int) BinaryCodec.MAX_USHORT;
+    }
+
+    private boolean isReferenceTooLargeForBAI() {
+        return header.getSequenceDictionary().getReferenceLength() > GenomicIndexUtil.BIN_GENOMIC_SPAN
+                && !isReferenceSizeWarningShowed;
```

Comment:
This works but it would be clearer to test `isReferenceSizeWarningShowed` at the same `if` block that sets it, instead of inside here. Also you should test the boolean first, if it's `true` then it doesn't matter if the reference length is too long or not.

Reply:
Done.